packages/grafana-ui/src/components/Layout/Box/Box.tsx

```diff
@@ -0,0 +1,214 @@
+  /** Sets the properties `padding-left` and `padding-right`. Higher priority than padding. */
+  paddingY?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-top`. Higher priority than padding and paddingY. */
+  paddingTop?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-bottom`. Higher priority than padding and paddingY. */
+  paddingBottom?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-left`. Higher priority than padding and paddingX. */
+  paddingLeft?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-right`. Higher priority than padding and paddingX. */
+  paddingRight?: ResponsiveProp<ThemeSpacingTokens>;
+
+  // Border Props
+  borderStyle?: ResponsiveProp<BorderStyle>;
+  borderColor?: ResponsiveProp<BorderColor>;
+  borderRadius?: ResponsiveProp<BorderRadius>;
+
+  // Flex Props
+  /** Sets the property `flex` */
+  grow?: ResponsiveProp<number>;
+  /** Sets the property `flex-shrink` */
+  shrink?: ResponsiveProp<number>;
+  alignItems?: ResponsiveProp<AlignItems>;
+  justifyContent?: ResponsiveProp<JustifyContent>;
+
+  // Other props
+  backgroundColor?: ResponsiveProp<BackgroundColor>;
+  display?: ResponsiveProp<Display>;
+  boxShadow?: ResponsiveProp<BoxShadow>;
+  /** Sets the HTML element that will be rendered as a Box. Defaults to 'div' */
+  element?: ElementType;
+}
+
+export const Box = ({ children, ...props }: React.PropsWithChildren<BoxProps>) => {
+  const styles = useStyles2(useCallback((theme) => getStyles(theme, props), [props]));
+  const Element = props.element ?? 'div';
+
+  return <Element className={styles.root}>{children}</Element>;
+};
+
+Box.displayName = 'Box';
+
+const customBorderColor = (color: BorderColor, theme: GrafanaTheme2): string | undefined => {
+  switch (color) {
+    case 'error':
+    case 'success':
+    case 'info':
+    case 'warning':
+      return theme.colors[color].border;
+    default:
+      return color ? theme.colors.border[color] : undefined;
+  }
+};
+
+const customBackgroundColor = (color: BackgroundColor, theme: GrafanaTheme2): string | undefined => {
+  switch (color) {
+    case 'error':
+    case 'success':
+    case 'info':
+    case 'warning':
+      return theme.colors[color].transparent;
+    default:
+      return color ? theme.colors.background[color] : undefined;
+  }
+};
+
+const getStyles = (theme: GrafanaTheme2, props: BoxProps) => {
+  const {
+    margin,
+    marginX,
+    marginY,
+    marginTop,
+    marginBottom,
+    marginLeft,
+    marginRight,
+    padding,
+    paddingX,
+    paddingY,
+    paddingTop,
+    paddingBottom,
+    paddingLeft,
+    paddingRight,
+    display,
+    backgroundColor,
+    grow,
+    shrink,
+    borderColor,
+    borderStyle,
+    borderRadius,
+    justifyContent,
+    alignItems,
+    boxShadow,
+  } = props;
+  return {
+    root: css([
+      getResponsiveStyle<ThemeSpacingTokens>(theme, margin, (val) => ({
```

Comment:
I don't think you need to specify the generic argument here? it will just be inferred from the values passed in? I tested removing it and it just works?

![image](https://github.com/grafana/grafana/assets/46142/aae3d60d-2974-4c8e-9826-267709f7d58e)